contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -318,6 +329,17 @@ func (h *requestOrResponseTrailerMapImpl) RangeWithCopy(f func(key, value string
 	}
 }
 
+func (h *requestOrResponseTrailerMapImpl) GetAllHeaders() map[string][]string {
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+	h.initTrailers()
+	copied_headers := make(map[string][]string)
+	for key, values := range h.headers {
+		copied_headers[key] = values
```

Comment:
this won't clone a slice too.

Reply:
I don't fully understand, can you elaborate?

Reply:
it is shallow copy now, here is an example:
```
	// slice
	s := []int{1, 2, 3}
	d := s
	s[0] = 4
	println(d[0]) // 4
```
I think we need deep clone for safety, thoughts?

Reply:
I'm not using this approach any more `d := s`

So I don't think your example applies.

````
map_one := map[string][]string{"one": {"a", "b", "c"}, "two": {"twoa", "twob", "twoc"}}

fmt.Printf("map_one: %v\n", map_one) \\ map_one: map[one:[a b c] two:[twoa twob twoc]]

map_copy := make(map[string][]string)

for key, value := range map_one {
    map_copy[key] = value
}

map_one["one"] = []string{"a1"}

fmt.Printf("map_copy: %v\n", map_copy) \\  map_copy: map[one:[a b c] two:[twoa twob twoc]]

fmt.Printf("map_copy key 'one': %v\n", map_copy["one"]) \\  map_copy key 'one': [a b c]

fmt.Printf("map_one key 'one': %v\n", map_one["one"]) \\  map_one key 'one': [a1]

```

Reply:
I see what you mean.... This should do what we need

```
	for key, value := range h.headers {
		copied_headers[key] = append(value[:0:0], value...)
	}
```

Or we could do this
```
	for key, value := range h.headers {
		copied_headers[key] = make([]string, len(value))
		copy(copied_headers[key], value)
	}

Reply:
The second option is more readable imo